tests/python/pants_test/reporting/test_reporting_integration.py

```diff
@@ -4,16 +4,25 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+import json
 import os.path
 import re
 import unittest
 from builtins import open
 
+from future.utils import PY2
 from parameterized import parameterized
 
+from pants.util.contextutil import http_server
 from pants_test.pants_run_integration_test import PantsRunIntegrationTest
 
 
+if PY2:
+  from BaseHTTPServer import BaseHTTPRequestHandler
+else:
+  from http.server import BaseHTTPRequestHandler
```

Comment:
See if this works without the Py2 import. Thanks to the future library, we should be able to just have the Py3 import and Py2 will use a backport.

Reply:
It works without Py2 import :) Thank you

Reply:
Next line worked in python 2:
`thrift_trace = self.rfile.read(int(self.headers.getheader('content-length')))`
But with python 3 I got the error:
`AttributeError: 'HTTPMessage' object has no attribute 'getheader'`
So I changed this line to:
`thrift_trace = self.rfile.read(int(self.headers.get('content-length')))`
Now it works for python 3. But I can not check if it now works for python 2.
I can do the check if it is python 2 or 3:
```
if PY2:
  thrift_trace = self.rfile.read(int(self.headers.getheader('content-length')))
else:
  thrift_trace = self.rfile.read(int(self.headers.get('content-length')))
```
Is it the only way? Could you please suggest if there is a better way.

Reply:
Hm I'm not sure and not able to test at the moment.

An acceptable solution would be
```python
content_length = self.headers.getheader('content-length') if PY3 else self.headers.get('content-length')
thrift_trace = self.rfile.read(int(content_length))
```

We put the PY3 branch in front with the idea of "Write Python 3 first code".